### PR TITLE
Fix RUNTIME_IDENTIFIER for accountingservice

### LINKED ISSUE #1798: accounting-service failing to start
# Bug Report

Which version of the demo you are using? ghcr.io/open-telemetry/demo:latest-accountingservice@sha256:a4c22f54468bf16fc32fc788e8f71d663b6886158847a5e555eb90381b6aae0e

## Symptom

The container is in a restart loop. The logs report:

```
Unhandled exception. System.IO.FileLoadException: Could not load file or assembly 'AccountingService, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
```

**What is the expected behavior?**

I would expect the container to reach a running state.

**What is the actual behavior?**

Container is in a restart loop.

## Reproduce

Run `make start`

We will close this issue if:

* The steps you provided are complex.
* If we can not reproduce the behavior you're reporting.

## Additional Context

I am attempting to run this on a Macbook Pro M2, so it would be pulling the arm64 versions.

### PR DESCRIPTION
**Pull request by pjanotti**
# Changes

Fix #1798 - the issue is that the .NET Docker images are Debian based, however, the runtime identifier (RID) passed on the build step is targeting `musl` based distros, e.g.: Alpine. Fixing the RID avoids the error mentioned on #1798.

### CODE REVIEW

File: src/cartservice/src/Dockerfile [hunk 1]
```diff
@@ -23,9 +23,9 @@ WORKDIR /usr/src/app/
 COPY ./src/cartservice/ ./
 COPY ./pb/ ./pb/
 
-RUN dotnet restore ./src/cartservice.csproj -r linux-musl-$TARGETARCH
```
[the remark this answers is not in the document]
Comment (julianocosta89) [reply src/cartservice/src/Dockerfile:26]: With that change the cartservice didn't start to me:

```
> docker logs cart-service 
exec ./cartservice: no such file or directory
```

**Comment by julianocosta89** [comment]
@pjanotti Thank you for the PR!
It made me understand better the issue.

In `cartservice` we build as `musl` because we run in Alpine: `mcr.microsoft.com/dotnet/runtime-deps:8.0-alpine3.20`.

For some reason, the `AccountingService`, when I switched to alpine it crashed. So I kept your change and also added the ARG `TARGETARCH` to the publish step.

@MMartyn would you be able to test this PR out?

**Comment by MMartyn** [comment]
> @MMartyn would you be able to test this PR out?
Tried it out and it works great. Thanks for the explanation on what was wrong.

**Review by julianocosta89 — APPROVED** [review]

**Comment by julianocosta89** [comment]
Thank you @pjanotti for fixing it, and thx @MMartyn for reporting and helping with the troubleshooting!

**Pull request merged**